src/main/java/com/github/firmwehr/gentle/firm/construction/FirmGraphBuilder.java

```diff
@@ -100,6 +114,23 @@ public void buildGraph(SProgram program) {
 		}
 	}
 
+	/**
+	 * This method wraps around the {@link Construction#newConst(int, Mode)} call to perform Const node caching. While
+	 * saving ressources, it's primary purpose is to reduce the cost of global value numbering in later stages. There
+	 * are still cases in which new Const nodes are created and one should not really on pointer comparisons, when
+	 * dealing with Const nodes.
+	 *
+	 * @param construction The construction instance in which the new Const node is to be created.
+	 * @param value The value of the constant.
+	 * @param mode The constant mode.
+	 *
+	 * @return A newly created Const node or an already created one from an earlier call.
+	 */
+	private Node newConst(Construction construction, int value, Mode mode) {
```

Comment:
This could probably moved into the `Context`. Then, the map is local to each graph and clearing isn't required anymore.

It would also passing the `Construction` to the method obsolete. WDYT?